src/js/stores/style.js

```diff
@@ -0,0 +1,115 @@
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Fluxxor = require("fluxxor");
+
+    var events = require("../events");
+
+    /**
+     * Manages a style clipboard and style link maps of active documents
+     */
+    var StyleStore = Fluxxor.createStore({
+        /**
+         * Currently stored style object for copy/pasting
+         * 
+         * @private
+         * @type {Object}
+         */
+        _storedStyle: null,
+
+        /** 
+         * Binds flux actions.
+         */
+        initialize: function () {
+            this._storedStyle = {
+                effects: {
+                    innerShadows: [],
+                    dropShadows: []
+                },
+                fillColor: null,
+                strokeColor: null,
+                typeStyle: null
+            };
+
+            this.bindActions(
+                events.style.COPY_STYLE, this._copyStyle
+            );
+        },
+
+        /**
+         * Returns the currently copied style object
+         *
+         * @return {Object}
+         */
+        getClipboardStyle: function () {
+            return this._storedStyle;
+        },
+
+        /**
+         * Copies the passed in layer's style to the clipboard
+         *
+         * @private
+         * @param {{layer: Layer}} payload
+         */
+        _copyStyle: function (payload) {
+            var layer = payload.layer,
+                fontStore = this.flux.store("font"),
+                fillColor = null,
+                strokeColor = null,
+                typeStyle = null;
+
+            switch (layer.kind) {
+            case layer.layerKinds.VECTOR:
+                fillColor = layer.fills.first() ? layer.fills.first().color : null;
+                strokeColor = layer.strokes.first() ? layer.strokes.first().color : null;
+
+                break;
+            case layer.layerKinds.TEXT:
+                fillColor = layer.text.characterStyle.color;
+                fillColor = fillColor ? fillColor.setOpacity(layer.opacity) : null;
+
+                strokeColor = null;
+                typeStyle = fontStore.getTypeObjectFromLayer(layer);
+
+                break;
```

Comment:
I'm surprised that JSHint doesn't complain about not having a default case. But I guess I don't care if it doesn't.